lua/toggleterm/terminal.lua

```diff
@@ -325,13 +327,31 @@ end
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stdout()
-  if self.on_stdout then return function(...) self.on_stdout(self, ...) end end
+  if self.auto_scroll or self.on_stdout then
+    return function(...)
+      if self.auto_scroll and vim.fn.bufexists(self.bufnr) == 1 then
+        vim.api.nvim_buf_call(self.bufnr, ui.scroll_to_bottom)
+      end
+      if self.on_stdout then
+        self.on_stdout(self, ...)
+      end
+    end
+  end
 end
 
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stderr()
-  if self.on_stderr then return function(...) self.on_stderr(self, ...) end end
+  if self.auto_scroll or self.on_stderr then
```

Comment:
```suggestion
  if self.on_stderr then
```
Same as above, only check for autoscroll in stderr not when deciding to define this function